Added the APTOS LedgerAccount test.

joaoccmartins opened the pull request:
<!--
Thank you for your contribution! 👍
Please make sure to read CONTRIBUTING.md if you have not already. Pull Requests that do not comply with the rules will be arbitrarily closed.
-->

### ✅ Checklist

<!-- Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready. -->

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** <!-- if not, please explain. (Feature must be tested / Bug fix must bring non-regression) -->
- [ ] **Impact of the changes:** <!-- Please take some time to list the impact & what specific areas Quality Assurance (QA) should focus on -->
  - ...

### 📝 Description

_Replace this text by a clear and concise description of what this pull request is about and why it is needed. Be sure to explain the problem you're addressing and the solution you're proposing._
_For libraries, you can add a code sample of how to use it._
_For bug fixes, you can explain the previous behaviour and how it was fixed._
_In case of visual features, please attach screenshots or video recordings to demonstrate the changes._

<!--
| Before        | After         |
| ------------- | ------------- |
|               |               |
-->

### ❓ Context

- **JIRA or GitHub link**: <!-- Attach the relevant ticket number if applicable. (e.g., [JIRA-123] for Jira or #123 for a Github issue) -->


---

### 🧐 Checklist for the PR Reviewers

<!-- Please do not edit this if you are the PR author -->

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

joaoccmartins commented on libs/ledger-live-common/src/families/aptos/LedgerAccount.test.ts:
```diff
@@ -1,37 +1,93 @@
+    });
+    HwAptos.prototype.signTransaction = mockSignTransaction;
+
+    await ledger_account.init(transport);
+    const payload = {
+      function: "0x1::aptos_account::transfer_coins",
+      typeArguments: [APTOS_ASSET_ID],
+      functionArguments: [Account.APTOS_1.address, BigInt(1).toString()],
+    };
+    const options = {
+      maxGasAmount: "100",
+      gasUnitPrice: "50",
+      sequenceNumber: "1",
+      expirationTimestampSecs: "1735639799486",
+    };
+
+    const rawTxn = await api.generateTransaction(Account.APTOS_1.address, payload, options);
+    const signedTxn = await ledger_account.signTransaction(rawTxn);
```
This is the portion that keeps failing. `signTransaction` is very picky about the mocked data, and I've tried multiple variations of generating it to no avail.

semeano reviewed (changes requested):
It's still not fully covered

<img width="791" alt="image" src="https://github.com/user-attachments/assets/03eb6ab5-eb61-45fe-9b85-b334880750c3" />

joaoccmartins commented:
> > It's still not fully covered
> > <img alt="image" width="791" src="https://private-user-images.githubusercontent.com/2123625/402905420-03eb6ab5-eb61-45fe-9b85-b334880750c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY4NTE5NTUsIm5iZiI6MTczNjg1MTY1NSwicGF0aCI6Ii8yMTIzNjI1LzQwMjkwNTQyMC0wM2ViNmFiNS1lYjYxLTQ1ZmUtOWI4NS1iMzM0ODgwNzUwYzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMTRUMTA0NzM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTJlZjE4ZjkyMGI0ZWRkODliOGJiYjk0NGM5MjZjNjE1MzZhMDY0ZTM0YTkxY2RkNWUwMjhmNTBhODU0MGM2OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.wV-jsn0K9gpLqMZxRqQFqqrBDZQTg5b199uBMDRGJPY">
> 
> I removed these functions, as they're not used anywhere. Might've brought them back on some merge. I'll remove them again.

Removed the functions again, and added exception testing to the asyncSignBuffer function."

may01 reviewed (commented):
Please add `npx changeset` as we merge to `develop`

joaoccmartins commented:
> Please add `npx changeset` as we merge to `develop`

Done.

semeano commented:
Tests are failing 😕
<img width="442" alt="image" src="https://github.com/user-attachments/assets/38e4524f-fbbe-4457-b1bd-783fa1f8e8bb" />
